sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -1255,6 +1255,13 @@ case class ExternalMapToCatalyst private(
   override def dataType: MapType = MapType(
     keyConverter.dataType, valueConverter.dataType, valueContainsNull = valueConverter.nullable)
 
+  private lazy val rowBuffer = InternalRow.fromSeq(Array[Any](1))
```

Comment:
We don't need to make this expression wide state. Just move this inside the `mapCatalystConverter` def.

Reply:
ok